{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b1e4af97",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "97457ee1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isolate</th>\n",
       "      <th>drug</th>\n",
       "      <th>confidence</th>\n",
       "      <th>gene</th>\n",
       "      <th>genome_index</th>\n",
       "      <th>GenTB Mutation</th>\n",
       "      <th>WHO Variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>inh</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>katG</td>\n",
       "      <td>2155168</td>\n",
       "      <td>SNP_CN_2155168_C944G_S315T_katG</td>\n",
       "      <td>katG_S315T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>rif</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>rpoB</td>\n",
       "      <td>761155</td>\n",
       "      <td>SNP_CN_761155_C1349T_S450L_rpoB</td>\n",
       "      <td>rpoB_S450L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>emb</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>embB</td>\n",
       "      <td>4248003</td>\n",
       "      <td>SNP_CN_4248003_A1490G_Q497R_embB</td>\n",
       "      <td>embB_Q497R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>amk</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>rrs</td>\n",
       "      <td>1473246</td>\n",
       "      <td>SNP_N_1473246_A1401G_rrs</td>\n",
       "      <td>rrs_a1401g</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>cap</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>rrs</td>\n",
       "      <td>1473246</td>\n",
       "      <td>SNP_N_1473246_A1401G_rrs</td>\n",
       "      <td>rrs_a1401g</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47235</th>\n",
       "      <td>TB-111-16552-18_S32_R.all</td>\n",
       "      <td>eth</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ethA</td>\n",
       "      <td>4326632</td>\n",
       "      <td>SNP_CN_4326632_T842G_H281P_ethA</td>\n",
       "      <td>ethA_H281P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47236</th>\n",
       "      <td>TB-111-17896-18_S97_R.all</td>\n",
       "      <td>str</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rpsL</td>\n",
       "      <td>781395</td>\n",
       "      <td>SNP_P_781395_T165C_promoter-rpsL</td>\n",
       "      <td>rpsL_t-165c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47237</th>\n",
       "      <td>TB-111-19509-18_S167_R.all</td>\n",
       "      <td>str</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rpsL</td>\n",
       "      <td>781395</td>\n",
       "      <td>SNP_P_781395_T165C_promoter-rpsL</td>\n",
       "      <td>rpsL_t-165c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47238</th>\n",
       "      <td>TB-111-18567-18-138_S44_R.all</td>\n",
       "      <td>str</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>gid</td>\n",
       "      <td>4407995</td>\n",
       "      <td>SNP_CN_4407995_T208G_S70R_gid</td>\n",
       "      <td>gid_S70R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47239</th>\n",
       "      <td>TB-111-18567-18-138_S44_R.all</td>\n",
       "      <td>str</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rpsL</td>\n",
       "      <td>781395</td>\n",
       "      <td>SNP_P_781395_T165C_promoter-rpsL</td>\n",
       "      <td>rpsL_t-165c</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47240 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             isolate drug                 confidence  gene  \\\n",
       "0                           TEST0248  inh               1) Assoc w R  katG   \n",
       "1                           TEST0248  rif               1) Assoc w R  rpoB   \n",
       "2                           TEST0248  emb               1) Assoc w R  embB   \n",
       "3                           TEST0248  amk               1) Assoc w R   rrs   \n",
       "4                           TEST0248  cap               1) Assoc w R   rrs   \n",
       "...                              ...  ...                        ...   ...   \n",
       "47235      TB-111-16552-18_S32_R.all  eth  3) Uncertain significance  ethA   \n",
       "47236      TB-111-17896-18_S97_R.all  str           5) Not assoc w R  rpsL   \n",
       "47237     TB-111-19509-18_S167_R.all  str           5) Not assoc w R  rpsL   \n",
       "47238  TB-111-18567-18-138_S44_R.all  str  3) Uncertain significance   gid   \n",
       "47239  TB-111-18567-18-138_S44_R.all  str           5) Not assoc w R  rpsL   \n",
       "\n",
       "       genome_index                    GenTB Mutation  WHO Variant  \n",
       "0           2155168   SNP_CN_2155168_C944G_S315T_katG   katG_S315T  \n",
       "1            761155   SNP_CN_761155_C1349T_S450L_rpoB   rpoB_S450L  \n",
       "2           4248003  SNP_CN_4248003_A1490G_Q497R_embB   embB_Q497R  \n",
       "3           1473246          SNP_N_1473246_A1401G_rrs   rrs_a1401g  \n",
       "4           1473246          SNP_N_1473246_A1401G_rrs   rrs_a1401g  \n",
       "...             ...                               ...          ...  \n",
       "47235       4326632   SNP_CN_4326632_T842G_H281P_ethA   ethA_H281P  \n",
       "47236        781395  SNP_P_781395_T165C_promoter-rpsL  rpsL_t-165c  \n",
       "47237        781395  SNP_P_781395_T165C_promoter-rpsL  rpsL_t-165c  \n",
       "47238       4407995     SNP_CN_4407995_T208G_S70R_gid     gid_S70R  \n",
       "47239        781395  SNP_P_781395_T165C_promoter-rpsL  rpsL_t-165c  \n",
       "\n",
       "[47240 rows x 7 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_mutations = pd.read_csv('/home/kin672/gentb-summer22/Mutations/all_mutations.csv')\n",
    "mutation_count = pd.read_csv('/home/kin672/gentb-summer22/Mutations/Mutation_Count.csv')\n",
    "full_db = pd.read_csv('/home/kin672/gentb-summer22/Creating Summaries from Relevant Files/7-18 Output/full_df_7-18.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7769ad95",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Summarize mutation identities for each isolate\n",
    "num_mutations_per_drug = {'Isolate':[], 'total':[], 'rif':[], 'inh':[], 'emb':[], 'pza':[], 'str':[], 'cap':[], 'amk':[], 'cip':[], 'kan':[], 'levo':[], 'oflx':[], 'pas':[], 'eth':[], 'unknown':[]}\n",
    "\n",
    "for isolate in full_db.ID.unique():\n",
    "    num_mutations_per_drug['Isolate'].append(isolate)\n",
    "    if isolate in mutation_count.Isolate.unique():\n",
    "        num_total_mutations = mutation_count.loc[mutation_count['Isolate'] == isolate, 'Num_Mutations'].values[0]\n",
    "        rif = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'rif')])\n",
    "        inh = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'inh')])\n",
    "        emb = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'emb')])\n",
    "        pza = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'pza')])\n",
    "        strep = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'str')])\n",
    "        cap = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'cap')])\n",
    "        amk = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'amk')])\n",
    "        cip = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'cip')])\n",
    "        kan = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'kan')])\n",
    "        levo = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'levo')])\n",
    "        oflx = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'oflx')])\n",
    "        pas = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'pas')])\n",
    "        eth = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'eth')])\n",
    "        not_specified = num_total_mutations - np.sum([rif, inh, emb, pza, strep, cap, amk, cip, kan, levo, oflx, pas, eth]) \n",
    "        total = num_total_mutations\n",
    "    else:\n",
    "        total, rif, inh, emb, pza, strep, cap, amk, cip, kan, levo, oflx, pas, eth, not_specified = 0\n",
    "    \n",
    "    num_mutations_per_drug['total'].append(total)\n",
    "    num_mutations_per_drug['rif'].append(rif)\n",
    "    num_mutations_per_drug['inh'].append(inh)\n",
    "    num_mutations_per_drug['emb'].append(emb)\n",
    "    num_mutations_per_drug['pza'].append(pza)\n",
    "    num_mutations_per_drug['str'].append(strep)\n",
    "    num_mutations_per_drug['cap'].append(cap)\n",
    "    num_mutations_per_drug['amk'].append(amk)\n",
    "    num_mutations_per_drug['cip'].append(cip)\n",
    "    num_mutations_per_drug['kan'].append(kan)\n",
    "    num_mutations_per_drug['levo'].append(levo)\n",
    "    num_mutations_per_drug['oflx'].append(oflx)\n",
    "    num_mutations_per_drug['pas'].append(pas)\n",
    "    num_mutations_per_drug['eth'].append(eth)\n",
    "    num_mutations_per_drug['unknown'].append(not_specified)\n",
    "    \n",
    "pd.DataFrame(num_mutations_per_drug).to_csv('/home/kin672/gentb-summer22/Mutations/Num_Mutations_Per_Drug.csv', index = False)\n",
    "\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "92dcf4b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function that produces confidence db for a single drug\n",
    "# Product db includes (for each isolate): Resistance, Lineage, # of WHO-identified mutations, # of high confidence mutations, # low confidence\n",
    "def summarize_mutations(drug, all_mutations = all_mutations, mutation_count = mutation_count):\n",
    "    db = {'Isolate':[], 'Lineage':[], 'Resistant':[], 'WHO-Identified':[], 'High_Confidence':[], 'Low_Confidence':[]}\n",
    "    \n",
    "    # for isolate in full_db.isolate.unique():\n",
    "    for isolate in ['TB-111-16552-18_S32_R.all']:\n",
    "        lineage = full_db.loc[(full_db['ID'] == isolate) & (full_db['Drug'] == 'rif'), 'Lineage'].values[0]\n",
    "        resistant = full_db.loc[(full_db['ID'] == isolate) & (full_db['Drug'] == 'rif'), 'Resistant'].values[0]\n",
    "        db['Isolate'].append(isolate)\n",
    "        db['Lineage'].append(lineage)\n",
    "        db['Resistant'].append(resistant)\n",
    "        \n",
    "        # If Isolate has any WHO-identified mutations, classify them\n",
    "        if isolate in all_mutations.isolate.unique():\n",
    "            WHO_mutations = all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == drug)]\n",
    "            num_WHO_mutations = len(WHO_mutations)\n",
    "            num_high_confidence = len(WHO_mutations[(WHO_mutations['confidence'] == '1) Assoc w R') | (WHO_mutations['confidence'] == '2) Assoc w R - Interim')])\n",
    "            num_low_confidence = num_WHO_mutations - num_high_confidence\n",
    "            \n",
    "            db['WHO-Identified'].append(num_WHO_mutations)\n",
    "            db['High_Confidence'].append(num_high_confidence)\n",
    "            db['Low_Confidence'].append(num_low_confidence)\n",
    "        else:\n",
    "            num_WHO_mutations, num_high_confidence, num_low_confidence = 0\n",
    "            db['WHO-Identified'].append(num_WHO_mutations)\n",
    "            db['High_Confidence'].append(num_high_confidence)\n",
    "            db['Low_Confidence'].append(num_low_confidence)\n",
    "            \n",
    "    return(pd.DataFrame(db))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21763174",
   "metadata": {},
   "outputs": [],
   "source": [
    "for drug in full_db.Drug.unique():\n",
    "    pd.DataFrame(summarize_mutations(drug)).to_csv('/home/kin672/gentb-summer22/Mutations/Breakdown Per Drug/' + str(drug) + '_summary_of_mutations.csv', index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "995fe0f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>syn</th>\n",
       "      <th>ntpos</th>\n",
       "      <th>coding</th>\n",
       "      <th>cref</th>\n",
       "      <th>cpos</th>\n",
       "      <th>cver</th>\n",
       "      <th>amino</th>\n",
       "      <th>aref</th>\n",
       "      <th>apos</th>\n",
       "      <th>aver</th>\n",
       "      <th>noncode</th>\n",
       "      <th>gene</th>\n",
       "      <th>rgene</th>\n",
       "      <th>Isolate</th>\n",
       "      <th>Full Mutation</th>\n",
       "      <th>mode</th>\n",
       "      <th>codes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CN</td>\n",
       "      <td>2155168</td>\n",
       "      <td>C944G</td>\n",
       "      <td>C</td>\n",
       "      <td>944</td>\n",
       "      <td>G</td>\n",
       "      <td>S315T</td>\n",
       "      <td>S</td>\n",
       "      <td>315.0</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>katG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TEST0248</td>\n",
       "      <td>SNP_CN_2155168_C944G_S315T_katG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CN</td>\n",
       "      <td>761155</td>\n",
       "      <td>C1349T</td>\n",
       "      <td>C</td>\n",
       "      <td>1349</td>\n",
       "      <td>T</td>\n",
       "      <td>S450L</td>\n",
       "      <td>S</td>\n",
       "      <td>450.0</td>\n",
       "      <td>L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TEST0248</td>\n",
       "      <td>SNP_CN_761155_C1349T_S450L_rpoB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CN</td>\n",
       "      <td>4248003</td>\n",
       "      <td>A1490G</td>\n",
       "      <td>A</td>\n",
       "      <td>1490</td>\n",
       "      <td>G</td>\n",
       "      <td>Q497R</td>\n",
       "      <td>Q</td>\n",
       "      <td>497.0</td>\n",
       "      <td>R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TEST0248</td>\n",
       "      <td>SNP_CN_4248003_A1490G_Q497R_embB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>N</td>\n",
       "      <td>1473246</td>\n",
       "      <td>A1401G</td>\n",
       "      <td>A</td>\n",
       "      <td>1401</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rrs</td>\n",
       "      <td>TEST0248</td>\n",
       "      <td>SNP_N_1473246_A1401G_rrs</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CN</td>\n",
       "      <td>7582</td>\n",
       "      <td>A281G</td>\n",
       "      <td>A</td>\n",
       "      <td>281</td>\n",
       "      <td>G</td>\n",
       "      <td>D94G</td>\n",
       "      <td>D</td>\n",
       "      <td>94.0</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TEST0248</td>\n",
       "      <td>SNP_CN_7582_A281G_D94G_gyrA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65255</th>\n",
       "      <td>N</td>\n",
       "      <td>1474001</td>\n",
       "      <td>C344T</td>\n",
       "      <td>C</td>\n",
       "      <td>344</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rrl</td>\n",
       "      <td>TB-111-19509-18_S167_R.all</td>\n",
       "      <td>SNP_N_1474001_C344T_rrl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65256</th>\n",
       "      <td>CN</td>\n",
       "      <td>4247145</td>\n",
       "      <td>C632T</td>\n",
       "      <td>C</td>\n",
       "      <td>632</td>\n",
       "      <td>T</td>\n",
       "      <td>S211F</td>\n",
       "      <td>S</td>\n",
       "      <td>211.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>embB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TB-111-19509-18_S167_R.all</td>\n",
       "      <td>SNP_CN_4247145_C632T_S211F_embB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65257</th>\n",
       "      <td>CN</td>\n",
       "      <td>4407995</td>\n",
       "      <td>T208G</td>\n",
       "      <td>T</td>\n",
       "      <td>208</td>\n",
       "      <td>G</td>\n",
       "      <td>S70R</td>\n",
       "      <td>S</td>\n",
       "      <td>70.0</td>\n",
       "      <td>R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TB-111-18567-18-138_S44_R.all</td>\n",
       "      <td>SNP_CN_4407995_T208G_S70R_gid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65258</th>\n",
       "      <td>P</td>\n",
       "      <td>781395</td>\n",
       "      <td>T165C</td>\n",
       "      <td>T</td>\n",
       "      <td>165</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>promoter</td>\n",
       "      <td>rpsL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TB-111-18567-18-138_S44_R.all</td>\n",
       "      <td>SNP_P_781395_T165C_promoter-rpsL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65259</th>\n",
       "      <td>N</td>\n",
       "      <td>1474001</td>\n",
       "      <td>C344T</td>\n",
       "      <td>C</td>\n",
       "      <td>344</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rrl</td>\n",
       "      <td>TB-111-18567-18-138_S44_R.all</td>\n",
       "      <td>SNP_N_1474001_C344T_rrl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65260 rows Ã— 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      syn    ntpos  coding cref  cpos cver  amino aref   apos aver   noncode  \\\n",
       "0      CN  2155168   C944G    C   944    G  S315T    S  315.0    T       NaN   \n",
       "1      CN   761155  C1349T    C  1349    T  S450L    S  450.0    L       NaN   \n",
       "2      CN  4248003  A1490G    A  1490    G  Q497R    Q  497.0    R       NaN   \n",
       "3       N  1473246  A1401G    A  1401    G    NaN  NaN    NaN  NaN       NaN   \n",
       "4      CN     7582   A281G    A   281    G   D94G    D   94.0    G       NaN   \n",
       "...    ..      ...     ...  ...   ...  ...    ...  ...    ...  ...       ...   \n",
       "65255   N  1474001   C344T    C   344    T    NaN  NaN    NaN  NaN       NaN   \n",
       "65256  CN  4247145   C632T    C   632    T  S211F    S  211.0    F       NaN   \n",
       "65257  CN  4407995   T208G    T   208    G   S70R    S   70.0    R       NaN   \n",
       "65258   P   781395   T165C    T   165    C    NaN  NaN    NaN  NaN  promoter   \n",
       "65259   N  1474001   C344T    C   344    T    NaN  NaN    NaN  NaN       NaN   \n",
       "\n",
       "       gene rgene                        Isolate  \\\n",
       "0      katG   NaN                       TEST0248   \n",
       "1      rpoB   NaN                       TEST0248   \n",
       "2      embB   NaN                       TEST0248   \n",
       "3       NaN   rrs                       TEST0248   \n",
       "4      gyrA   NaN                       TEST0248   \n",
       "...     ...   ...                            ...   \n",
       "65255   NaN   rrl     TB-111-19509-18_S167_R.all   \n",
       "65256  embB   NaN     TB-111-19509-18_S167_R.all   \n",
       "65257   gid   NaN  TB-111-18567-18-138_S44_R.all   \n",
       "65258  rpsL   NaN  TB-111-18567-18-138_S44_R.all   \n",
       "65259   NaN   rrl  TB-111-18567-18-138_S44_R.all   \n",
       "\n",
       "                          Full Mutation mode codes  \n",
       "0       SNP_CN_2155168_C944G_S315T_katG  NaN   NaN  \n",
       "1       SNP_CN_761155_C1349T_S450L_rpoB  NaN   NaN  \n",
       "2      SNP_CN_4248003_A1490G_Q497R_embB  NaN   NaN  \n",
       "3              SNP_N_1473246_A1401G_rrs  NaN   NaN  \n",
       "4           SNP_CN_7582_A281G_D94G_gyrA  NaN   NaN  \n",
       "...                                 ...  ...   ...  \n",
       "65255           SNP_N_1474001_C344T_rrl  NaN   NaN  \n",
       "65256   SNP_CN_4247145_C632T_S211F_embB  NaN   NaN  \n",
       "65257     SNP_CN_4407995_T208G_S70R_gid  NaN   NaN  \n",
       "65258  SNP_P_781395_T165C_promoter-rpsL  NaN   NaN  \n",
       "65259           SNP_N_1474001_C344T_rrl  NaN   NaN  \n",
       "\n",
       "[65260 rows x 17 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('/home/kin672/gentb-summer22/Mutations/Mutation_Info_Martin.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "48da72eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>drug</th>\n",
       "      <th>genome_index</th>\n",
       "      <th>confidence</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>AMI</td>\n",
       "      <td>4268928</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>aftB</td>\n",
       "      <td>aftB_c-92t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>AMI</td>\n",
       "      <td>4268964</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>aftB</td>\n",
       "      <td>aftB_g-128t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>AMI</td>\n",
       "      <td>619747</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-144a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>AMI</td>\n",
       "      <td>619700</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-191t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>AMI</td>\n",
       "      <td>619795</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-96g</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17528</th>\n",
       "      <td>17567</td>\n",
       "      <td>STM</td>\n",
       "      <td>3568779</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB7</td>\n",
       "      <td>whiB7_t-100c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17529</th>\n",
       "      <td>17568</td>\n",
       "      <td>STM</td>\n",
       "      <td>4407620</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>gid</td>\n",
       "      <td>gid_Y195H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17530</th>\n",
       "      <td>17569</td>\n",
       "      <td>STM</td>\n",
       "      <td>3568488</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB7</td>\n",
       "      <td>whiB7_192_del_1_gc_g</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17531</th>\n",
       "      <td>17570</td>\n",
       "      <td>STM</td>\n",
       "      <td>1472337</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rrs</td>\n",
       "      <td>rrs_c492t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17532</th>\n",
       "      <td>17572</td>\n",
       "      <td>STM</td>\n",
       "      <td>4338589</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB6</td>\n",
       "      <td>whiB6_-68_del_1_ctccgagc_cgccgag</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17533 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0 drug genome_index                 confidence   gene  \\\n",
       "0               0  AMI      4268928  3) Uncertain significance   aftB   \n",
       "1               1  AMI      4268964  3) Uncertain significance   aftB   \n",
       "2               2  AMI       619747  3) Uncertain significance   ccsA   \n",
       "3               3  AMI       619700  3) Uncertain significance   ccsA   \n",
       "4               4  AMI       619795  3) Uncertain significance   ccsA   \n",
       "...           ...  ...          ...                        ...    ...   \n",
       "17528       17567  STM      3568779           5) Not assoc w R  whiB7   \n",
       "17529       17568  STM      4407620           5) Not assoc w R    gid   \n",
       "17530       17569  STM      3568488           5) Not assoc w R  whiB7   \n",
       "17531       17570  STM      1472337           5) Not assoc w R    rrs   \n",
       "17532       17572  STM      4338589           5) Not assoc w R  whiB6   \n",
       "\n",
       "                                variant  \n",
       "0                            aftB_c-92t  \n",
       "1                           aftB_g-128t  \n",
       "2                           ccsA_c-144a  \n",
       "3                           ccsA_c-191t  \n",
       "4                            ccsA_c-96g  \n",
       "...                                 ...  \n",
       "17528                      whiB7_t-100c  \n",
       "17529                         gid_Y195H  \n",
       "17530              whiB7_192_del_1_gc_g  \n",
       "17531                         rrs_c492t  \n",
       "17532  whiB6_-68_del_1_ctccgagc_cgccgag  \n",
       "\n",
       "[17533 rows x 6 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('/n/data1/hms/dbmi/farhat/Sanjana/MIC_data/WHO_resistance_variants_all.csv')\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "jupytervenv",
   "language": "python",
   "name": "jupytervenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
