{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6a47e643",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json\n",
    "from pandas.api.types import CategoricalDtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3377a5d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>drug</th>\n",
       "      <th>genome_index</th>\n",
       "      <th>confidence</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>AMI</td>\n",
       "      <td>4268928</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>aftB</td>\n",
       "      <td>aftB_c-92t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>AMI</td>\n",
       "      <td>4268964</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>aftB</td>\n",
       "      <td>aftB_g-128t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>AMI</td>\n",
       "      <td>619747</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-144a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>AMI</td>\n",
       "      <td>619700</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-191t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>AMI</td>\n",
       "      <td>619795</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-96g</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17528</th>\n",
       "      <td>17567</td>\n",
       "      <td>STM</td>\n",
       "      <td>3568779</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB7</td>\n",
       "      <td>whiB7_t-100c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17529</th>\n",
       "      <td>17568</td>\n",
       "      <td>STM</td>\n",
       "      <td>4407620</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>gid</td>\n",
       "      <td>gid_Y195H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17530</th>\n",
       "      <td>17569</td>\n",
       "      <td>STM</td>\n",
       "      <td>3568488</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB7</td>\n",
       "      <td>whiB7_192_del_1_gc_g</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17531</th>\n",
       "      <td>17570</td>\n",
       "      <td>STM</td>\n",
       "      <td>1472337</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rrs</td>\n",
       "      <td>rrs_c492t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17532</th>\n",
       "      <td>17572</td>\n",
       "      <td>STM</td>\n",
       "      <td>4338589</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB6</td>\n",
       "      <td>whiB6_-68_del_1_ctccgagc_cgccgag</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17533 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0 drug genome_index                 confidence   gene  \\\n",
       "0               0  AMI      4268928  3) Uncertain significance   aftB   \n",
       "1               1  AMI      4268964  3) Uncertain significance   aftB   \n",
       "2               2  AMI       619747  3) Uncertain significance   ccsA   \n",
       "3               3  AMI       619700  3) Uncertain significance   ccsA   \n",
       "4               4  AMI       619795  3) Uncertain significance   ccsA   \n",
       "...           ...  ...          ...                        ...    ...   \n",
       "17528       17567  STM      3568779           5) Not assoc w R  whiB7   \n",
       "17529       17568  STM      4407620           5) Not assoc w R    gid   \n",
       "17530       17569  STM      3568488           5) Not assoc w R  whiB7   \n",
       "17531       17570  STM      1472337           5) Not assoc w R    rrs   \n",
       "17532       17572  STM      4338589           5) Not assoc w R  whiB6   \n",
       "\n",
       "                                variant  \n",
       "0                            aftB_c-92t  \n",
       "1                           aftB_g-128t  \n",
       "2                           ccsA_c-144a  \n",
       "3                           ccsA_c-191t  \n",
       "4                            ccsA_c-96g  \n",
       "...                                 ...  \n",
       "17528                      whiB7_t-100c  \n",
       "17529                         gid_Y195H  \n",
       "17530              whiB7_192_del_1_gc_g  \n",
       "17531                         rrs_c492t  \n",
       "17532  whiB6_-68_del_1_ctccgagc_cgccgag  \n",
       "\n",
       "[17533 rows x 6 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutations = pd.read_csv('/n/data1/hms/dbmi/farhat/Sanjana/MIC_data/WHO_resistance_variants_all.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "534702f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Switching the abbreviations to the ones that I've been using, and removing any that I haven't been working with\n",
    "drug_translation = {'AMI': 'amk', 'BDQ': 'bdq', 'CAP':'cap', 'CFZ':'cfz', 'DLM':'dlm', 'EMB':'emb', 'ETH':'eth', 'INH':'inh', 'KAN':'kan',\n",
    "       'LZD':'lzd', 'PZA':'pza', 'RIF':'rif', 'STM':'str', 'LEV':'levo', 'MXF': 'moxi'}\n",
    "mutations.drug = [drug_translation[original] for original in mutations.drug]\n",
    "mutations = mutations.loc[[drug in ['amk', 'cap', 'emb', 'eth', 'inh', 'kan', 'rif', 'pza', 'str', 'levo'] for drug in mutations.drug],:]\n",
    "mutations['searchable_variant'] = [str(i).replace('-','').upper() for i in list(mutations.variant)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9d885179",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('//home//kin672//gentb-summer22//Isolating Relevant Files//7-18 Output//relevant_file_names_7-18.json') as f:\n",
    "    relevant_files = json.load(f) # This is a dictionary with format {STRAIN ID: {Folder: , Resistance: , Lineage: }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "22ace0a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def break_down(string):\n",
    "    broken = {'original': string, 'gene': [], '1': [], '2': [], 'genome_index': [], 'change': []}\n",
    "    temp = string.split('_')\n",
    "    broken['gene'] = temp.pop(-1)\n",
    "    broken['1'] = temp.pop(0)\n",
    "    broken['2'] = temp.pop(0)\n",
    "    broken['genome_index'] = temp.pop(0)\n",
    "    broken['change'] = temp\n",
    "\n",
    "    return(broken)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "d59689d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_db = pd.DataFrame({'isolate':[], 'drug':[], 'confidence':[], 'gene':[], 'genome_index':[], 'GenTB Mutation':[], 'WHO Variant':[]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "810f1e6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>drug</th>\n",
       "      <th>genome_index</th>\n",
       "      <th>confidence</th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>searchable_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>amk</td>\n",
       "      <td>4268928</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>aftB</td>\n",
       "      <td>aftB_c-92t</td>\n",
       "      <td>AFTB_C92T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>amk</td>\n",
       "      <td>4268964</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>aftB</td>\n",
       "      <td>aftB_g-128t</td>\n",
       "      <td>AFTB_G128T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>amk</td>\n",
       "      <td>619747</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-144a</td>\n",
       "      <td>CCSA_C144A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>amk</td>\n",
       "      <td>619700</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-191t</td>\n",
       "      <td>CCSA_C191T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>amk</td>\n",
       "      <td>619795</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ccsA</td>\n",
       "      <td>ccsA_c-96g</td>\n",
       "      <td>CCSA_C96G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17528</th>\n",
       "      <td>17567</td>\n",
       "      <td>str</td>\n",
       "      <td>3568779</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB7</td>\n",
       "      <td>whiB7_t-100c</td>\n",
       "      <td>WHIB7_T100C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17529</th>\n",
       "      <td>17568</td>\n",
       "      <td>str</td>\n",
       "      <td>4407620</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>gid</td>\n",
       "      <td>gid_Y195H</td>\n",
       "      <td>GID_Y195H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17530</th>\n",
       "      <td>17569</td>\n",
       "      <td>str</td>\n",
       "      <td>3568488</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB7</td>\n",
       "      <td>whiB7_192_del_1_gc_g</td>\n",
       "      <td>WHIB7_192_DEL_1_GC_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17531</th>\n",
       "      <td>17570</td>\n",
       "      <td>str</td>\n",
       "      <td>1472337</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rrs</td>\n",
       "      <td>rrs_c492t</td>\n",
       "      <td>RRS_C492T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17532</th>\n",
       "      <td>17572</td>\n",
       "      <td>str</td>\n",
       "      <td>4338589</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>whiB6</td>\n",
       "      <td>whiB6_-68_del_1_ctccgagc_cgccgag</td>\n",
       "      <td>WHIB6_68_DEL_1_CTCCGAGC_CGCCGAG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15029 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0 drug genome_index                 confidence   gene  \\\n",
       "0               0  amk      4268928  3) Uncertain significance   aftB   \n",
       "1               1  amk      4268964  3) Uncertain significance   aftB   \n",
       "2               2  amk       619747  3) Uncertain significance   ccsA   \n",
       "3               3  amk       619700  3) Uncertain significance   ccsA   \n",
       "4               4  amk       619795  3) Uncertain significance   ccsA   \n",
       "...           ...  ...          ...                        ...    ...   \n",
       "17528       17567  str      3568779           5) Not assoc w R  whiB7   \n",
       "17529       17568  str      4407620           5) Not assoc w R    gid   \n",
       "17530       17569  str      3568488           5) Not assoc w R  whiB7   \n",
       "17531       17570  str      1472337           5) Not assoc w R    rrs   \n",
       "17532       17572  str      4338589           5) Not assoc w R  whiB6   \n",
       "\n",
       "                                variant               searchable_variant  \n",
       "0                            aftB_c-92t                        AFTB_C92T  \n",
       "1                           aftB_g-128t                       AFTB_G128T  \n",
       "2                           ccsA_c-144a                       CCSA_C144A  \n",
       "3                           ccsA_c-191t                       CCSA_C191T  \n",
       "4                            ccsA_c-96g                        CCSA_C96G  \n",
       "...                                 ...                              ...  \n",
       "17528                      whiB7_t-100c                      WHIB7_T100C  \n",
       "17529                         gid_Y195H                        GID_Y195H  \n",
       "17530              whiB7_192_del_1_gc_g             WHIB7_192_DEL_1_GC_G  \n",
       "17531                         rrs_c492t                        RRS_C492T  \n",
       "17532  whiB6_-68_del_1_ctccgagc_cgccgag  WHIB6_68_DEL_1_CTCCGAGC_CGCCGAG  \n",
       "\n",
       "[15029 rows x 7 columns]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prefix = '//n//groups//gentb_www//predictData//'\n",
    "\n",
    "# Loop through every unique isolate\n",
    "for strainID in relevant_files:\n",
    "    value = relevant_files.get(strainID)\n",
    "    json_path = prefix + value.get('Folder') + '//' + value.get('Resistance')\n",
    "    \n",
    "    # Parse json\n",
    "    with open(json_path) as f:\n",
    "        resistance = json.load(f) \n",
    "    \n",
    "    # Get full list of resistance-related mutations for this isolate\n",
    "    mut_list = [i for sublist in resistance[1].values() for sublist2 in sublist for i in sublist2 if i != None]\n",
    "    mut_list = mut_list + [i for sublist in resistance[2].values() for sublist2 in sublist for i in sublist2 if i != None]\n",
    "    mut_list = mut_list + [item for sublist in resistance[3].values() for item in sublist if item != 'Null']\n",
    "    mut_list = list(pd.Series(mut_list).unique())\n",
    "    \n",
    "    if len(mut_list) == 0:\n",
    "        continue\n",
    "    \n",
    "    # Match the mutation with the corresponding mutation(s) in the WHO database\n",
    "    for mut in mut_list:\n",
    "        x = break_down(mut)\n",
    "        candidates = mutations.loc[mutations.genome_index == x['genome_index'],:]\n",
    "        \n",
    "        # Make sure the gene aligns\n",
    "        candidates = candidates.loc[[gene in x['gene'] for gene in candidates.gene], :]\n",
    "        \n",
    "        # Isolate the candidates who have one of the amino acid swaps indicated\n",
    "        candidates = candidates.loc[[any(code in searchable_variant for code in x['change']) for searchable_variant in candidates.searchable_variant], :]  \n",
    "        \n",
    "        # Move on if there are no relevant mutations\n",
    "        if len(candidates) == 0:\n",
    "            continue\n",
    "            \n",
    "        temp = candidates.loc[:,['drug', 'genome_index', 'confidence', 'gene']]\n",
    "        temp['WHO Variant'] = candidates.variant\n",
    "        temp['isolate'] = strainID\n",
    "        temp['GenTB Mutation'] = x['original']\n",
    "        \n",
    "        final_db = pd.concat([final_db, temp], ignore_index = True)\n",
    "            \n",
    "        \n",
    "        \n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "966fe005",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (596733944.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Input \u001b[0;32mIn [190]\u001b[0;36m\u001b[0m\n\u001b[0;31m    final_db.to_csv('/home/kin672/gentb-summer22/Mutations/all_mutations.csv', index = )\u001b[0m\n\u001b[0m                                                                                       ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "final_db.to_csv('/home/kin672/gentb-summer22/Mutations/all_mutations.csv', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "jupytervenv",
   "language": "python",
   "name": "jupytervenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
