{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b1e4af97",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "97457ee1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isolate</th>\n",
       "      <th>drug</th>\n",
       "      <th>confidence</th>\n",
       "      <th>gene</th>\n",
       "      <th>genome_index</th>\n",
       "      <th>GenTB Mutation</th>\n",
       "      <th>WHO Variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>inh</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>katG</td>\n",
       "      <td>2155168</td>\n",
       "      <td>SNP_CN_2155168_C944G_S315T_katG</td>\n",
       "      <td>katG_S315T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>rif</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>rpoB</td>\n",
       "      <td>761155</td>\n",
       "      <td>SNP_CN_761155_C1349T_S450L_rpoB</td>\n",
       "      <td>rpoB_S450L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>emb</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>embB</td>\n",
       "      <td>4248003</td>\n",
       "      <td>SNP_CN_4248003_A1490G_Q497R_embB</td>\n",
       "      <td>embB_Q497R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>amk</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>rrs</td>\n",
       "      <td>1473246</td>\n",
       "      <td>SNP_N_1473246_A1401G_rrs</td>\n",
       "      <td>rrs_a1401g</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TEST0248</td>\n",
       "      <td>cap</td>\n",
       "      <td>1) Assoc w R</td>\n",
       "      <td>rrs</td>\n",
       "      <td>1473246</td>\n",
       "      <td>SNP_N_1473246_A1401G_rrs</td>\n",
       "      <td>rrs_a1401g</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47235</th>\n",
       "      <td>TB-111-16552-18_S32_R.all</td>\n",
       "      <td>eth</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>ethA</td>\n",
       "      <td>4326632</td>\n",
       "      <td>SNP_CN_4326632_T842G_H281P_ethA</td>\n",
       "      <td>ethA_H281P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47236</th>\n",
       "      <td>TB-111-17896-18_S97_R.all</td>\n",
       "      <td>str</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rpsL</td>\n",
       "      <td>781395</td>\n",
       "      <td>SNP_P_781395_T165C_promoter-rpsL</td>\n",
       "      <td>rpsL_t-165c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47237</th>\n",
       "      <td>TB-111-19509-18_S167_R.all</td>\n",
       "      <td>str</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rpsL</td>\n",
       "      <td>781395</td>\n",
       "      <td>SNP_P_781395_T165C_promoter-rpsL</td>\n",
       "      <td>rpsL_t-165c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47238</th>\n",
       "      <td>TB-111-18567-18-138_S44_R.all</td>\n",
       "      <td>str</td>\n",
       "      <td>3) Uncertain significance</td>\n",
       "      <td>gid</td>\n",
       "      <td>4407995</td>\n",
       "      <td>SNP_CN_4407995_T208G_S70R_gid</td>\n",
       "      <td>gid_S70R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47239</th>\n",
       "      <td>TB-111-18567-18-138_S44_R.all</td>\n",
       "      <td>str</td>\n",
       "      <td>5) Not assoc w R</td>\n",
       "      <td>rpsL</td>\n",
       "      <td>781395</td>\n",
       "      <td>SNP_P_781395_T165C_promoter-rpsL</td>\n",
       "      <td>rpsL_t-165c</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47240 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             isolate drug                 confidence  gene  \\\n",
       "0                           TEST0248  inh               1) Assoc w R  katG   \n",
       "1                           TEST0248  rif               1) Assoc w R  rpoB   \n",
       "2                           TEST0248  emb               1) Assoc w R  embB   \n",
       "3                           TEST0248  amk               1) Assoc w R   rrs   \n",
       "4                           TEST0248  cap               1) Assoc w R   rrs   \n",
       "...                              ...  ...                        ...   ...   \n",
       "47235      TB-111-16552-18_S32_R.all  eth  3) Uncertain significance  ethA   \n",
       "47236      TB-111-17896-18_S97_R.all  str           5) Not assoc w R  rpsL   \n",
       "47237     TB-111-19509-18_S167_R.all  str           5) Not assoc w R  rpsL   \n",
       "47238  TB-111-18567-18-138_S44_R.all  str  3) Uncertain significance   gid   \n",
       "47239  TB-111-18567-18-138_S44_R.all  str           5) Not assoc w R  rpsL   \n",
       "\n",
       "       genome_index                    GenTB Mutation  WHO Variant  \n",
       "0           2155168   SNP_CN_2155168_C944G_S315T_katG   katG_S315T  \n",
       "1            761155   SNP_CN_761155_C1349T_S450L_rpoB   rpoB_S450L  \n",
       "2           4248003  SNP_CN_4248003_A1490G_Q497R_embB   embB_Q497R  \n",
       "3           1473246          SNP_N_1473246_A1401G_rrs   rrs_a1401g  \n",
       "4           1473246          SNP_N_1473246_A1401G_rrs   rrs_a1401g  \n",
       "...             ...                               ...          ...  \n",
       "47235       4326632   SNP_CN_4326632_T842G_H281P_ethA   ethA_H281P  \n",
       "47236        781395  SNP_P_781395_T165C_promoter-rpsL  rpsL_t-165c  \n",
       "47237        781395  SNP_P_781395_T165C_promoter-rpsL  rpsL_t-165c  \n",
       "47238       4407995     SNP_CN_4407995_T208G_S70R_gid     gid_S70R  \n",
       "47239        781395  SNP_P_781395_T165C_promoter-rpsL  rpsL_t-165c  \n",
       "\n",
       "[47240 rows x 7 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_mutations = pd.read_csv('/home/kin672/gentb-summer22/Mutations/all_mutations.csv')\n",
    "mutation_count = pd.read_csv('/home/kin672/gentb-summer22/Mutations/Mutation_Count.csv')\n",
    "full_db = pd.read_csv('/home/kin672/gentb-summer22/Creating Summaries from Relevant Files/7-18 Output/full_df_7-18.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7769ad95",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Summarize mutation identities for each isolate\n",
    "num_mutations_per_drug = {'Isolate':[], 'total':[], 'rif':[], 'inh':[], 'emb':[], 'pza':[], 'str':[], 'cap':[], 'amk':[], 'cip':[], 'kan':[], 'levo':[], 'oflx':[], 'pas':[], 'eth':[], 'unknown':[]}\n",
    "\n",
    "for isolate in full_db.ID.unique():\n",
    "    num_mutations_per_drug['Isolate'].append(isolate)\n",
    "    if isolate in mutation_count.Isolate.unique():\n",
    "        num_total_mutations = mutation_count.loc[mutation_count['Isolate'] == isolate, 'Num_Mutations'].values[0]\n",
    "        rif = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'rif')])\n",
    "        inh = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'inh')])\n",
    "        emb = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'emb')])\n",
    "        pza = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'pza')])\n",
    "        strep = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'str')])\n",
    "        cap = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'cap')])\n",
    "        amk = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'amk')])\n",
    "        cip = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'cip')])\n",
    "        kan = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'kan')])\n",
    "        levo = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'levo')])\n",
    "        oflx = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'oflx')])\n",
    "        pas = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'pas')])\n",
    "        eth = len(all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == 'eth')])\n",
    "        not_specified = num_total_mutations - np.sum([rif, inh, emb, pza, strep, cap, amk, cip, kan, levo, oflx, pas, eth]) \n",
    "        total = num_total_mutations\n",
    "    else:\n",
    "        total, rif, inh, emb, pza, strep, cap, amk, cip, kan, levo, oflx, pas, eth, not_specified = 0\n",
    "    \n",
    "    num_mutations_per_drug['total'].append(total)\n",
    "    num_mutations_per_drug['rif'].append(rif)\n",
    "    num_mutations_per_drug['inh'].append(inh)\n",
    "    num_mutations_per_drug['emb'].append(emb)\n",
    "    num_mutations_per_drug['pza'].append(pza)\n",
    "    num_mutations_per_drug['str'].append(strep)\n",
    "    num_mutations_per_drug['cap'].append(cap)\n",
    "    num_mutations_per_drug['amk'].append(amk)\n",
    "    num_mutations_per_drug['cip'].append(cip)\n",
    "    num_mutations_per_drug['kan'].append(kan)\n",
    "    num_mutations_per_drug['levo'].append(levo)\n",
    "    num_mutations_per_drug['oflx'].append(oflx)\n",
    "    num_mutations_per_drug['pas'].append(pas)\n",
    "    num_mutations_per_drug['eth'].append(eth)\n",
    "    num_mutations_per_drug['unknown'].append(not_specified)\n",
    "    \n",
    "pd.DataFrame(num_mutations_per_drug).to_csv('/home/kin672/gentb-summer22/Mutations/Num_Mutations_Per_Drug.csv', index = False)\n",
    "\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "92dcf4b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function that produces confidence db for a single drug\n",
    "# Product db includes (for each isolate): Resistance, Lineage, # of WHO-identified mutations, # of high confidence mutations, # low confidence\n",
    "def summarize_mutations(drug, all_mutations = all_mutations, mutation_count = mutation_count):\n",
    "    db = {'Isolate':[], 'Lineage':[], 'Resistant':[], 'WHO-Identified':[], 'High_Confidence':[], 'Low_Confidence':[]}\n",
    "    \n",
    "    # for isolate in full_db.isolate.unique():\n",
    "    for isolate in ['TB-111-16552-18_S32_R.all']:\n",
    "        lineage = full_db.loc[(full_db['ID'] == isolate) & (full_db['Drug'] == 'rif'), 'Lineage'].values[0]\n",
    "        resistant = full_db.loc[(full_db['ID'] == isolate) & (full_db['Drug'] == 'rif'), 'Resistant'].values[0]\n",
    "        db['Isolate'].append(isolate)\n",
    "        db['Lineage'].append(lineage)\n",
    "        db['Resistant'].append(resistant)\n",
    "        \n",
    "        # If Isolate has any WHO-identified mutations, classify them\n",
    "        if isolate in all_mutations.isolate.unique():\n",
    "            WHO_mutations = all_mutations[(all_mutations['isolate'] == isolate) & (all_mutations['drug'] == drug)]\n",
    "            num_WHO_mutations = len(WHO_mutations)\n",
    "            num_high_confidence = len(WHO_mutations[(WHO_mutations['confidence'] == '1) Assoc w R') | (WHO_mutations['confidence'] == '2) Assoc w R - Interim')])\n",
    "            num_low_confidence = num_WHO_mutations - num_high_confidence\n",
    "            \n",
    "            db['WHO-Identified'].append(num_WHO_mutations)\n",
    "            db['High_Confidence'].append(num_high_confidence)\n",
    "            db['Low_Confidence'].append(num_low_confidence)\n",
    "        else:\n",
    "            num_WHO_mutations, num_high_confidence, num_low_confidence = 0\n",
    "            db['WHO-Identified'].append(num_WHO_mutations)\n",
    "            db['High_Confidence'].append(num_high_confidence)\n",
    "            db['Low_Confidence'].append(num_low_confidence)\n",
    "            \n",
    "    return(pd.DataFrame(db))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21763174",
   "metadata": {},
   "outputs": [],
   "source": [
    "for drug in full_db.Drug.unique():\n",
    "    pd.DataFrame(summarize_mutations(drug)).to_csv('/home/kin672/gentb-summer22/Mutations/Breakdown Per Drug/' + str(drug) + '_summary_of_mutations.csv', index = False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "jupytervenv",
   "language": "python",
   "name": "jupytervenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
